src/mock/parsec/mod.rs

```diff
@@ -221,11 +221,31 @@ where
 
     pub fn has_unpolled_observations(&self) -> bool {
         state::with::<T, S::PublicId, _, _>(self.section_hash, |state| {
-            state.has_unconsensused_observations(&self.peer_list)
+            state
+                .unconsensused_observations_for_peers(&self.peer_list)
+                .next()
+                .is_some()
                 || state.get_block(self.first_unconsensused).is_some()
         }) || self.our_unpolled_observations().next().is_some()
     }
 
+    pub fn unpolled_observations_string(&self) -> String {
+        let value = state::with::<T, S::PublicId, _, _>(self.section_hash, |state| {
+            format!(
+                "unconsensused_observations_for_peers: {:?}, first_unconsensused block: {:?}",
```

Comment:
is the info of the `peers` needed?

Reply:
I think that is useful, but not sure we need the "first unconsensused block". Isn't that going to be included among the unconsensused observations anyway?

Reply:
No, one is the first unconsused blok by any peer, while `our_unpolled_observations` are only the one we voted for ourselves.

Reply:
My question was about whether we need `first_unconsensused_block`, not whether we need `out_unpolled_observations`. The later one is definitely useful, but the former I think is already included in `unconsensused_observations_for_peers`, so might be redundant.

Reply:
Is it, then why did we have that condition, was it redundant?
```
             state.has_unconsensused_observations(&self.peer_list)
                || state.get_block(self.first_unconsensused).is_some() 
```

Reply:
Is it not a consensused observation if it is a block, just happen to not have been polled yet?

Reply:
Hm, good point. OK, let's leave it in.

Reply:
Yeah, I think `state.get_block(self.first_unconsensused)` is a block that already reached consensus but this node hasn't realized it yet. `has_unconsensused_observations` is for observations that haven't reached consensus yet. So it's fine as is.